plugins/unreal/XJGame/Plugins/XJMusicPlugin/Source/XJMusicPlugin/Private/XjMusicInstanceSubsystem.cpp

```diff
@@ -0,0 +1,188 @@
+// Fill out your copyright notice in the Description page of Project Settings.
+
+
+#include "XjMusicInstanceSubsystem.h"
+#include "Kismet/GameplayStatics.h"
+#include <Settings/XJMusicDefaultSettings.h>
+#include <Sound/SoundBase.h>
+#include <Sound/SoundWave.h>
+#include <Engine/StreamableManager.h>
+#include <Engine/AssetManager.h>
+#include <Engine/ObjectLibrary.h>
+#include <Misc/FileHelper.h>
+#include <Runtime/Engine/Public/AudioDevice.h>
+
+#include "TimerManager.h"
+
+void UXjMusicInstanceSubsystem::RetrieveProjectsContent(const FString& Directory)
+{
+	WorldAudioDeviceHandle = GetWorld()->GetAudioDevice();
+
+	TArray<FString> WavFiles;
+
+	UXJMusicDefaultSettings* XjSettings = GetMutableDefault<UXJMusicDefaultSettings>();
+	if (!XjSettings)
+	{
+		return;
+	}
+
+	FString WorkPath = Directory;
+
+	IPlatformFile& PlatformFile = FPlatformFileManager::Get().GetPlatformFile();
+
+	if (!PlatformFile.DirectoryExists(*WorkPath))
+	{
+		return;
+	}
+
+	TArray<FString> FoundAudioFilesPaths;
+
+	PlatformFile.FindFilesRecursively(FoundAudioFilesPaths, *WorkPath, *AudioExtension);
```

Comment:
This assumes the audio file extension, which is fine for now. But it made me think, this whole RetrieveProjectContents function could instead wait for the content to be loaded, then check the content for allInstrumentAudios and then load them, instead of finding files on disk.